controllers/eventbus/controller.go

```diff
@@ -57,14 +57,20 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	if reconcileErr != nil {
 		log.Errorw("reconcile error", zap.Error(reconcileErr))
 	}
+
+	// client.Update() mutates busCopy so we need to make an
```

Comment:
In the call to `r.client.Update(ctx, busCopy)`, the `busCopy` variable is a pointer and can be mutated by the updated function, resulting in an unexpected object being passed to `r.client.Status().Update(ctx, busCopy)`. This is a bug we discovered while doing this PR.

All this is probably moot now since there is a [separate PR](https://github.com/argoproj/argo-events/pull/2466) to address this bug.